src/runtime_src/core/pcie/tools/cloud-daemon/azure/azure.cpp

```diff
@@ -233,6 +233,13 @@ int AzureDev::azureLoadXclBin(const xclBin *buffer)
     if (fpgaSerialNumber.empty())
         return -E_EMPTY_SN;
     std::cout << "LoadXclBin FPGA serial No: " << fpgaSerialNumber << std::endl;
+
+    // check if the xclbin is valid
+    if (xclbin::get_axlf_section(buffer, BITSTREAM) != nullptr) {
+        std::cout << "xclbin is invalid, please provide azure xclbin" << std::endl;
+        return -EINVAL;
```

Comment:
I would suggest we define azure specific errno, like E_INVALID_XCLBIN.
EINVAL is also used in xocl driver, sometimes when we see this errno, we don't know it fails in xocl or plugin.

Reply:
What is an azure specific error code?  Any errors propagated to end user should be exception (if C++) or POSIX error code if (C).   If the error ends up in a syslog some place, the error code shouldn't matter, should it?

Reply:
the errno in the log should clearly tell what is wrong. The practice in azure plugin is, any failure in the plugin returns a specific defined error code, instead of the POSIX errno that are also used in xocl driver, so that, we can identify which part of the plugin fails by examining the log

Reply:
Got it.  I see the error codes defined in azure_rest_err.